diff --git a/main.py b/main.py
index 3ff2475..a67e7ab 100644
--- a/main.py
+++ b/main.py
@@ -3,15 +3,20 @@
 import sys
 import time
 import urllib
-from os import getenv
+from os import path, getenv
 from telebot import TeleBot, types
 
+if getenv("CREDENTIALS_DIRECTORY"):
+    print("using bot token provided by init")
+    with open(path.join(getenv("CREDENTIALS_DIRECTORY"), "bot-token"), "r") as token:
+        Bot_TOKEN = token.readline()
+else:
+    Bot_TOKEN = getenv("arch_man_bot_token")
 
-Bot_TOKEN = getenv("arch_man_bot_token")
 if Bot_TOKEN is None:
     sys.exit("You must set <arch_man_bot_token> environment variable!")
 
-bot = TeleBot(Bot_TOKEN)
+bot = TeleBot(Bot_TOKEN.rstrip("\n"))
 shouldBotLog = True  # if False bot will not create and keep log.log file
 
 
